### DISCUSSION TITLE
Network data consumption Using CodeSpaces in VSCode

### DISCUSSION THREAD

**Discussion by goyalyashpal** [community/community#18833] [Codespaces]
* For normal coding, i.e. just creating source files, how much network data does it consume say per hour?
* When a session is disconnected (session_expired), then on reconnecting, does it download all of it all over again, or just the changes compared to local cache?
* Can it run offline?
* How can i make it to consume less network data?

**Comment by dmgardiner25** [comment]
@yashpalgoyal1304 
1. If you open the command palette in VS Code (ctrl+shift+p or F1) and run the "Preferences: Open Settings (UI), you can enable the "Github > Codespaces: Show Performance Explorer" setting which will allow you to see information about your codespace, including the total network data (in and out) during your session.
<img width="600" alt="image" src="https://user-images.githubusercontent.com/22037769/174672352-25420597-31a4-49e1-9c68-1367f3f83702.png">
<img width="400" alt="image" src="https://user-images.githubusercontent.com/22037769/174672755-2100e575-5fe3-4c32-8f75-0a8104716fe6.png">

2. I'm not sure what all gets cached (as I believe this would be handled by VS Code), maybe @osortega has an idea?
3. We don't currently have support for codespace running offline.
4. Are you seeing your codespace consuming large amounts of data? Most data usage should be from file operations as internal codespace operations are very small and shouldn't be contributing much to the total data usage.

**Reply by goyalyashpal** [reply]
1. Thanks a lot for this
2. awaiting input
3. Ohkay
4. No, but 
    * i was seeing the commands in the terminal to be very slow, like it was lagging a LOT behind in handling the commands, and 
    * it was disconnecting too often

----------
Tangentially, and lightly asking
... like you have hidden that first field in "Github Codespaces" how do i know which info is to be kept hidden? I mean the two hidden fields in the "Codespace Performance" i get - they say "ID", but that one topmost field doesnt say any such thing.

**Reply by goyalyashpal** [reply]
> 4\. Are you seeing your codespace consuming large amounts of data?

hey @dmgardiner25 so, 
* the task of rebuilding container take huge amount of data - 300-400 MiB of it - so, i was borrowing the data from someone, and now the per is frowning upon me :(
* And this all happens with an innocent looking notification popup with single click - WITHOUT any confirmation nor intimation of this much data required in any form, 
* atleast the huge size should shown on the progress bar of the notification popup.

what more, it says "rebuild container?" and even the fact that it will require a download is not mentioned 😢. so, it - due to large size download - stalled at the progress bar in that notification popup - and i exited VS code thinking there was some other problem - this might have already costed me few 100 megs more of the data

**minor update:** 
* for now, i've set the codespaces extension setting of `github.codespaces.devcontainerChangedNotificationStyle` to `none` from default of `notification` 
* but i do hope the raised points gets looked into

**Reply by goyalyashpal** [reply]
i have seen this behaviour of downloading huge amount of data without any information pan MS products, be it :
* windows updates, 
* downloading something via MS store, (no, the shown size is not the true size of download)
* using github desktop, 
* updating MS powertoys, (yes, this did not told that it was downloading either)
* or now github codespaces 

- informing about the required download data size is not paid any attention to at all 
- okay, granted u people are rich and don't care about few hundreds of MB or even Gigs of data
- but people somewhere may have their financial constraints 😢😢😢

don't take the last second point personally, i am just tired of wrestling with this behaviour, 
i just wanna if a process is requiring internet connection or is just operating on existing local data (i.e. extracting, copy pasting etc)
and if former, then how much data is required 
so that i can plan according to it, 
is it too much to ask for?

**Reply by prem40008** [reply]
Too costly to make projects on codespace saara internet khatam , tata bye-bye
